tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5794,6 +5812,8 @@ class HLSLExternalSource : public ExternalSemaSource {
     return method;
   }
 
+  CallGraphWithRecurseGuard getCallGraph() { return m_callGraph; }
```

Comment:
This should *definitely* return a reference, not a copy.  There could be const version too, if that's necessary.